core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/StructuredStreamingPlugin.scala

```
@@ -0,0 +1,109 @@
+
+            case `_: FileStreamWriter`(fileStreamWriter) =>
+              val fileStreamSink = extractFieldValue[FileStreamSink](fileStreamWriter, "fileStreamSink")
+              val path = extractFieldValue[String](fileStreamSink, "path")
+              val fileFormat = extractFieldValue[DataSourceRegister](fileStreamSink, "fileFormat")
+              val options = extractFieldValue[Map[String, String]](fileStreamSink, "options")
+
+              (SourceIdentifier(Some(fileFormat.shortName()), asURI(path)), SaveMode.Append, query, options)
+          }
+      }
+  }
+}
+
+object StructuredStreamingPlugin {
+  object `_: StreamingDataSourceV2Relation` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.execution.datasources.v2.StreamingDataSourceV2Relation")
+
+  object `_: KafkaMicroBatchReader` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.kafka010.KafkaMicroBatchReader")
+
+  object `_: KafkaStreamWriter` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.kafka010.KafkaStreamWriter")
+
+  object `_: WriteToDataSourceV2` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2")
+
+  object `_: MicroBatchWriter` extends SafeTypeMatchingExtractor[AnyRef](
+    "org.apache.spark.sql.execution.streaming.sources.MicroBatchWriter")
+
+  object `_: FileStreamWriter` extends SafeTypeMatchingExtractor[AnyRef](
+    "za.co.absa.spline.harvester.plugin.embedded.FileStreamWriter")
+
+  private def asURI(topic: String) = s"kafka:$topic"
+}
+
+// https://issues.apache.org/jira/browse/SPARK-27484
+class FileStreamWriter(val fileStreamSink: FileStreamSink) extends StreamWriter {
+  override def commit(epochId: Long, messages: Array[WriterCommitMessage]): Unit = {}
+
+  override def abort(epochId: Long, messages: Array[WriterCommitMessage]): Unit = {}
+
+  override def createWriterFactory(): DataWriterFactory[InternalRow] = null
```

How about throwing `UnsupportedOperationException` instead of returning `null`?

---

Yop

---

ok